docs/reasons.md

```diff
@@ -9,122 +9,122 @@ By understanding these failure reasons, operators can better monitor, troublesho
+  - *Summary*: bug: partial signature database didn`t trigger partial signature exchange, this is unexpected
+  - *Details*: Reason `bug_par_sig_db_internal` indicates a bug in the partial signature database as it is unexpected. Note this may happen due to expiry race.
+
+### Failure Reason: `bug_sig_agg`
+  - *Summary*: bug: threshold aggregation of partial signatures failed due to inconsistent signed data
+  - *Details*: Reason `bug_sig_agg` indicates that BLS threshold aggregation of sufficient partial signatures failed. This indicates inconsistent signed data. This indicates a bug in charon as it is unexpected.
 
-### Failure Reason: `fetcher_aggregator_failed_prepare`
+### Failure Reason: `failed_aggregator_selection`
   - *Summary*: couldn`t aggregate attestation due to failed prepare aggregator duty
-  - *Details*: Reason `fetcher_aggregator_failed_prepare` indicates an attestation aggregation duty failed in the fetcher step since it couldn`t fetch the prerequisite aggregated beacon committee selections. This indicates the associated prepare aggregation duty failed.
+  - *Details*: Reason `failed_aggregator_selection` indicates an attestation aggregation duty failed in the fetcher step since it couldn`t fetch the prerequisite aggregated beacon committee selections. This indicates the associated prepare aggregation duty failed.
 
-### Failure Reason: `fetcher_aggregator_few_prepares`
+### Failure Reason: `failed_proposer_randao`
+  - *Summary*: couldn`t propose block due to failed randao duty
+  - *Details*: Reason `failed_proposer_randao` indicates a block proposer duty failed in the fetcher step since it couldn`t fetch the prerequisite aggregated RANDAO. This indicates the associated randao duty failed.
+
+### Failure Reason: `fetch_bn_error`
+  - *Summary*: couldn`t fetch duty data from the beacon node
+  - *Details*: Reason `fetch_bn_error` indicates a duty failed in the fetcher step when it failed to fetch the required data from the beacon node API. This indicates a problem with the upstream beacon node.
+
+### Failure Reason: `insufficient_aggregator_selections`
   - *Summary*: couldn`t aggregate attestation due to insufficient partial beacon committee selections
-  - *Details*: Reason `fetcher_aggregator_few_prepares` indicates an attestation aggregation duty failed in the fetcher step since it couldn`t fetch the prerequisite aggregated beacon committee selections. This indicates the associated prepare aggregation duty failed due to insufficient partial beacon committee selections submitted by the cluster validator clients.
+  - *Details*: Reason `insufficient_aggregator_selections` indicates an attestation aggregation duty failed in the fetcher step since it couldn`t fetch the prerequisite aggregated beacon committee selections. This indicates the associated prepare aggregation duty failed due to insufficient partial beacon committee selections submitted by the cluster validator clients.
```

Comment:
```suggestion
  - *Details*: Reason `insufficient_aggregator_selections` indicates an attestation aggregation duty failed in the fetcher step since it couldn't fetch the prerequisite aggregated beacon committee selections. This indicates the associated prepare aggregation duty failed due to insufficient partial beacon committee selections submitted by the cluster validator clients.
```

There might be more instances of back ticks used as apostrophes, it will break Markdown parsing.

Reply:
`reasons.md` is generated from reason.go. I cannot patch it manually as it will break the test.

Reply:
Interesting that the source contains `couldn't`

Reply:
Found it: `result := strings.ReplaceAll(buf.String(), "'", "`")`
https://github.com/ObolNetwork/charon/blob/main/cmd/markdown_internal_test.go#L195

Shall we remove this you think?

Reply:
<img width="1058" alt="image" src="https://github.com/ObolNetwork/charon/assets/6468078/d044acb0-eba6-47e0-a2a3-3df867947528">

it does not seem break the markdown.

Reply:
We're lucky then! Backticks are used for inline code, weird that it doesn't break.

I'm approving then!

Reply:
I believe it looks for the second backtick on the same line. This is not triggered in this case.
Also, note that in our test, we explicitly replace all ' with ` here https://github.com/ObolNetwork/charon/blob/main/cmd/markdown_internal_test.go#L195
The source text (reason.go) contains '